### DISCUSSION TITLE
Enable/disable certain field types

### DISCUSSION THREAD

**Discussion by geertvansoest** [solspace/craft-freeform#405] [Feature Requests]
Hi,

At first, I have to say we are working with Freeform a lot! 😄 

But at the most of the websites we build for our clients, we don't need that huge amount of field types. Especially when we use the Pro version, there are a lot of types we don't want to use, like:
- Rating
- Opinion scale
- Signature
- Table
- Etc.

But since the most of our clients build their forms theirselves using the form builder, they are able to use those fields anyway (although they rarely do). And in that case, it's our responsibility for all those fields to look properly and nice at all dimensions and devices.

Because of that we're wondering if it's possible to add a feature to enable/disable certain form fields (simply don't show them in the form builder in the admin panel). 

Kind regard,
Geert van Soest

**Comment by kjmartens** [comment]
Thanks for your feedback, @geertvansoest. We do have plans to add support for this in the not-too-distant future. 🙂

**Comment by thisisjamessmith** [comment]
Related to this discussion, (and also related to my [earlier post](https://github.com/solspace/craft-freeform/discussions/477) about fully disabling Solspace's JavaScript), I'd really like the ability to optionally remove all of the settings in the Form Settings sidebar of the form builder, as well as being able to remove the power to create multiple pages and arrange fields into columns (and optionally limit the number of columns they can create when that ability is enabled).

Most of my clients have no idea what AJAX is (and neither should they!). They also should not have the power to select their own formatting template (we build one and one only! If they want more they need to pay extra!). Likewise for the Google Tag Manager integration - we have existing systems to handle all goal completions and tracking events in a unified way - the last thing we want is for clients to be able to manage that on a per-form basis themselves. Things like the "Show Loading Indicator on Submit" on the Validation tab, I would also argue, should almost never be left in the hands of a client - that shouldn't be their decision to make!

All these things, along with the plethora of seldom-used fieldtypes mentioned above make it look to the client as though the plugin is broken or unreliable because they can see functionality that may or may not have been configured, when the truth is that doing a full implementation of every piece of Solspace's functionality could take just as long as building a small website!

**It's great that you have all these features, but we need to be able to bill a client for each one separately.**

I realise a lot of this functionality comes down to the expectation that developers are automatically including Solspace's bundled JavaScript, but I'm really not keen on allowing third-party front end code to be automatically output into any of our sites - it's why we don't use Wordpress, after all. Craft, and EE before it, have always been 'bring-your-own-front-end'-flavoured systems, and personally I'd like to see Freeform move closer to that same philosophy.

Sorry I don't mean to sound all ranty - Freeform is the "best" form plugin available in my opinion (though I like Express Forms a lot), it just feels like the feature set and associated complexity has gotten somewhat out of hand... Which is fine as long as it's configurable!

**Reply by kjmartens** [reply]
Thanks for your feedback, @thisisjamessmith. I can assure you that we are currently addressing these issues you mention in our next larger update to Freeform.

**Comment by jeffreyschmitz87** [comment]
Do you have an update regarding enabling/disabling certain field types?

**Reply by kjmartens** [reply]
We will have an update in the coming months... likely by late Spring 2023. 🙂

**Comment by kjmartens** [comment]
This has been implemented in Freeform 5 (now in beta). 🙂

**Comment by kjmartens** [comment]
Additionally, Freeform 5.4 introduces a new "Limited Users" permission layer to assign to users or user groups. Within that, you can prevent users from using specific field types. 🙂
